feat(YouTube/Settings): Remove quotations from proper nouns

KobeW50 opened the pull request:
Closes https://github.com/inotia00/ReVanced_Extended/issues/2131

inotia00's TODO (after PR is merged?):

- [x]  Update string name for `revanced_hide_subscriptions_channel_section` to `revanced_hide_subscriptions_carousel`

- [x] Update string name for `revanced_return_shorts_channel_name` to `revanced_replace_channel_handle`

KobeW50 committed:
YTM: Remove quotations from proper nouns

KobeW50 committed:
YT: Remove quotations from proper nouns

KobeW50 commented on src/main/resources/youtube/settings/host/values/strings.xml:
```diff
@@ -494,7 +494,7 @@ Tap and hold to open RVX settings."</string>
     <string name="revanced_disable_speed_overlay_summary">"Disable 'Playing at 2x speed' while holding down.
```
I don't know how to change `Disable 'Playing at 2x speed' while holding down.`.

For me, the overlay just says `2x>>` instead of `Playing at 2x speed`. 

Does it make sense to rename this option to something like `Disable hold to 2x speed`? This emphasizes that this option doesn't just hide an overlay but disables this action entirely.

inotia00 replied:
When this was first added to YouTube by Google, the `Playing at 2x speed` label was used. (Added from YouTube 18.19.36)
At some point, it seems to have been changed to `2x>>` label.

KobeW50 commented:
I'm unsure whether this YT string should have the quotes kept:

`To open YouTube links in RVX, enable \'Open supported links\' and enable the supported web addresses.`

ILoveOpenSourceApplications commented:
Also, in comments, `Hide create Shorts button` should be changed to `Hide Create a Short button`.

KobeW50 commented:
> Also, in comments, `Hide create Shorts button` should be changed to `Hide Create a Short button`.

Hmm, I'm not sure if this is correct. As far as I'm aware, "Create a Short" is the title of the page after pressing the button, but the button doesn't have a label.

I'll wait for inotia00's input. 

(Also, if we keep the current string, should it be changed to `Hide create Short button`?)

KobeW50 commented:
> Also, in comments, `Hide create Shorts button` should be changed to `Hide Create a Short button`.

Hmm, I'm not sure if this is correct. As far as I'm aware, "Create a Short" is the title of the page after pressing the button, but the button doesn't have a label.

I'll wait for inotia00's input.

ILoveOpenSourceApplications commented:
And in description, instead of section everything is called sections. Similarly, rename podcast section to Explore the podcast section, capitalize first word of Chapters and Transcript sections. And change suggestions sections description to `Featured places, Games and Music sections are hidden/shown.`

Also @KobeW50, can you look into this [comment](https://github.com/inotia00/revanced-patches/pull/54#issuecomment-2150389199), this [comment](https://github.com/inotia00/revanced-patches/pull/54#issuecomment-2150422012) and maybe this [comment](https://github.com/inotia00/revanced-patches/pull/53#issuecomment-2134479081)?

ILoveOpenSourceApplications commented:
> Hmm, I'm not sure if this is correct. As far as I'm aware, "Create a Short" is the title of the page after pressing the button, but the button doesn't have a label.

Yes, the button itself doesn't have a label. But the litho used is `composer_short_creation_button`. Otherwise it should be called, literally as `Hide camera button`.

inotia00 commented:
> I'm unsure whether this YT string should have the quotes kept:
> 
> `To open YouTube links in RVX, enable \'Open supported links\' and enable the supported web addresses.`

It is better to refrain from using the quote as much as possible, but I think it would be better to allow quotations in these part

> Also, in comments, `Hide create Shorts button` should be changed to `Hide Create a Short button`.

According to [docs](https://support.google.com/youtube/answer/10059070?hl=en#zippy=%2Chow-do-i-create-shorts%2Ccan-i-earn-money-from-my-shorts) and strings.xml, it seems to be written as `Create a Short`

docs:
![Capture](https://github.com/inotia00/revanced-patches/assets/108592928/86bc4d8f-359d-46c2-a9bb-e40eecb9125b)

strings.xml:
```
    <string name="shorts_camera_allow_access_title">Create a Short</string>
```

KobeW50 commented:
> ... And change suggestions sections description to `Featured places, Games and Music sections are hidden/shown.`

Should the setting title be renamed to `Hide attributes sections`?

> Also @KobeW50, can you look into this [comment](https://github.com/inotia00/revanced-patches/pull/54#issuecomment-2150389199), this [comment](https://github.com/inotia00/revanced-patches/pull/54#issuecomment-2150422012) and maybe this [comment](https://github.com/inotia00/revanced-patches/pull/53#issuecomment-2134479081)?

I'll implement the first one and the second part of the second one. For the 3rd one, I don't know how to reorder the settings (and I don't have the patience to).

ILoveOpenSourceApplications commented:
> Should the setting title be renamed to `Hide attributes sections`?

Would be better than suggestions.

KobeW50 committed:
Apply changes from review

KobeW50 committed:
Update strings.xml

ILoveOpenSourceApplications commented:
There's also `Append time stamp information` to be changed as well.

KobeW50 committed:
Update strings.xml

KobeW50 committed:
Speed overlay setting description

ILoveOpenSourceApplications commented:
Maybe rename `Return Shorts channel name` to `Replace channel handle`?

ILoveOpenSourceApplications commented:
Options with `disable` should also have similar descriptions like `enable`. Currently, `Disable ambient mode` description is 'Disables ambient mode` whereas it should've been Ambient mode is enabled and Ambient mode is disabled when toggled off and on respectively. Multiple such corrections are required here and there.

ILoveOpenSourceApplications commented on src/main/resources/youtube/settings/host/values/strings.xml:
```diff
@@ -922,24 +922,24 @@ Please download %2$s from the website."</string>
-    <string name="revanced_hide_transcript_section_title">Hide transcript sections</string>
-    <string name="revanced_hide_transcript_section_summary_on">Transcript sections are hidden.</string>
-    <string name="revanced_hide_transcript_section_summary_off">Transcript sections are shown.</string>
+    <string name="revanced_hide_suggestions_section_title">Hide attributes sections</string>
```
Shouldn't it be attributes section as a video will only have any one of the section and not more than that?

KobeW50 replied:
Ah, I didn't realize that. Thanks

KobeW50 committed:
Update strings.xml

ILoveOpenSourceApplications commented:
`Hide subscriptions channel section` sounds off. Can we phrase it to something better?

KobeW50 commented:
> `Hide subscriptions channel section` sounds off. Can we phrase it to something better?

Maybe `Hide subscriptions carousel`? @inotia00 what do u think?

Also, I need to update the revanced_pref.xml if the string name (and setting name)  is changed?

KobeW50 commented:
> Options with `disable` should also have similar descriptions like `enable`. Currently, `Disable ambient mode` description is 'Disables ambient mode` whereas it should've been Ambient mode is enabled and Ambient mode is disabled when toggled off and on respectively. Multiple such corrections are required here and there.

Unfortunately, I don't have time to do this atm. I only opened this PR bec I knew fixing the proper nouns would be quick and simple. Please open a PR or issue for this.

KobeW50 committed:
Rename toggle

inotia00 commented:
> Maybe `Hide subscriptions carousel`? @inotia00 what do u think?

Yeap looks good

> Also, I need to update the revanced_pref.xml if the string name (and setting name) is changed?

It is ideal to do so, but if so, it also requires updates on integrations and crowdin (because there is a change in the setting key)

I will manually change this part after the PR is merged later

KobeW50 commented:
> > Also, I need to update the revanced_pref.xml if the string name (and setting name) is changed?
> 
> It is ideal to do so, but if so, it also requires updates on integrations and crowdin (because there is a change in the setting key)
> 
> I will manually change this part after the PR is merged later
> 

In https://github.com/inotia00/revanced-patches/commit/00054435fb4f9b4b87d18548cb54751e1d63a0cb I updated a string name (even though I wasn't too sure what I was doing.) Should I revert this and leave it to you to deal with later?

inotia00 commented:
> In [0005443](https://github.com/inotia00/revanced-patches/commit/00054435fb4f9b4b87d18548cb54751e1d63a0cb) I updated a string name (even though I wasn't too sure what I was doing. Should I revert this and leave it to you to deal with later?

If you are planning to leave PR in Integrations, you need to change the following value:

https://github.com/inotia00/revanced-integrations/blob/18d776ce2c53492978746b8bbb099a3d5faef200/app/src/main/java/app/revanced/integrations/youtube/settings/Settings.java#L381

```
public static final BooleanSetting RETURN_SHORTS_CHANNEL_NAME = new BooleanSetting("revanced_return_shorts_channel_name", FALSE);
```
to
```
public static final BooleanSetting RETURN_SHORTS_CHANNEL_NAME = new BooleanSetting("revanced_replace_channel_handle", FALSE);
```
(because the value of `key` has been changed)


If you have no plan to open a PR in integrations, string name should be rolled back

KobeW50 committed:
Revert string name change

This partially reverts commit 00054435fb4f9b4b87d18548cb54751e1d63a0cb.

KobeW50 committed:
Apply suggestion from review (Hide subscriptions channel section)

KobeW50 committed:
Apply suggestion from review

KobeW50 committed:
Unfuck my forced commit mess

(Also made another description dynamic)

KobeW50 commented:
I think this is ready.

Some notes: 

1. [This suggestion](https://github.com/inotia00/revanced-patches/pull/56#issuecomment-2179098630) was not fully resolved, because there are still other examples of this issue outside the `Ambient mode` settings.

2. I messed up inotia00's commit where he added a string for when `Disable ambient mode` is On and Off. (IDK how GH works 🤷). I think I corrected my error in my most recent commit. In my recent commit, I also added a string for when `Disable ambient mode in fullscreen` is On and Off. IDK if there are now integrations and crowdin chores to do.

3. I added a small checklist for inotia00 to do in the description of this PR. Edit it as you see fit.

ILoveOpenSourceApplications commented on src/main/resources/youtube/settings/host/values/strings.xml:
```diff
@@ -591,9 +591,11 @@ Settings → Autoplay → Autoplay next video"</string>
+    <string name="revanced_disable_ambient_mode_summary_off">Ambient mode is enabled.</string>
     <string name="revanced_disable_ambient_mode_in_fullscreen_title">Disable ambient mode in fullscreen</string>
-    <string name="revanced_disable_ambient_mode_in_fullscreen_summary">Disables ambient mode for fullscreen only.</string>
+    <string name="revanced_disable_ambient_mode_in_fullscreen_summary_on">Ambient mode is only disabled for fullscreen.</string>
```
Isn't `Ambient mode is disabled/enabled in fullscreen` enough when toggle is on/off?

KobeW50 replied:
Then there would be a case where the `Disable ambient mode` toggle is enabled and the `Disable ambient mode in fullscreen description` would still say "Ambient mode is enabled in fullscreen."

This implies that Ambient is currently working in fullscreen, but in reality it is disabled everywhere.

Ideally, there are two completely independent toggles:

1. `Disable ambient mode under player`
2. `Disable ambient mode in fullscreen`

ILoveOpenSourceApplications replied:
> This implies that Ambient is currently working in fullscreen, but in reality it is disabled everywhere.

Does it matter though, as the option gets grayed out, if `Disable Ambient mode` is toggled on. 

Also, there's other options which has similar description style. An example being the `Hide Shorts shelves`.

ILoveOpenSourceApplications commented:
> * [ ]   Update string name for `revanced_hide_subscriptions_channel_section` to `revanced_hide_subscriptions_channel_carousel`

Edit: It should be `revanced_hide_subscriptions_carousel`.

KobeW50 committed:
Apply suggestion from review

ILoveOpenSourceApplications commented:
Now you have to change the option to `Disable Ambient mode for fullscreen` and the on toggle `Ambient mode is disabled for fullscreen` as well. 
~Also, shouldn't the A in ambient be capitalized?~

KobeW50 commented:
> the on toggle `Ambient mode is disabled for fullscreen` as well.

It's better to specify "only". Lots of dynamic descriptions aren't pure opposites for the on/off positions.

ILoveOpenSourceApplications commented:
Doesn't it already specify fullscreen, so is `only` actually required? Given there are only two places ambient mode is currently available. 

Edit: Once again, there are other options with similar description styles.

KobeW50 commented:
> Doesn't it already specify fullscreen, so is `only` actually required? Given there are only two places ambient mode is currently available.

True. But descriptions are meant to simplify things for users, and here "only" makes it explicit without making the sentence lengthy. 

Also, "only" is useful because it implies that the `Disable ambient mode` will disable all ambient mode.


The string also had "only" before this PR so I think it should be kept.

KobeW50 committed:
capitalize `Ambient mode` and change wording

ILoveOpenSourceApplications commented:
> Also, "only" is useful because it implies that the `Disable ambient mode` will disable all ambient mode.
> 
> The string also had "only" before this PR so I think it should be kept.

Now here's an example. Here also "only in home feed." could be used. But it isn't necessary as there's only one home feed. The case is same here. There's only one fullscreen (whether portrait/landscape as the toggle affects both), it isn't necessary to specify "only in fullscreen" here either.

![Screenshot_20240620_235355_YouTube](https://github.com/inotia00/revanced-patches/assets/117499019/f1382456-bb00-47ac-9b79-961da9ad1ed0)

Also @inotia00, does toggling on `Hide Comments section` hide it in home feed as well or not?

ILoveOpenSourceApplications commented:
Btw, I'm not sure whether "Ambient mode" is considered a noun. When checking https://support.google.com/youtube/answer/12827017, they have written it in running letters. So it may not need a capital A, which is why I canceled my initial statement.

Edit: Mainly because other options from the video flyout menu isn't capitalized. An example being "playback speed".

inotia00 commented:
> Also @inotia00, does toggling on `Hide Comments section` hide it in home feed as well or not?

When I checked [integrations](https://github.com/inotia00/revanced-integrations/blob/18d776ce2c53492978746b8bbb099a3d5faef200/app/src/main/java/app/revanced/integrations/youtube/patches/components/CommentsFilter.java#L108
)...
The first setting only hides the comments section in the player, the second setting only hides the comments section in the home feed

inotia00 commented:
> Btw, I'm not sure whether "Ambient mode" is considered a noun. When checking https://support.google.com/youtube/answer/12827017, they have written it in running letters. So it may not need a capital A, which is why I canceled my initial statement.
> 
> Edit: Mainly because other options from the video flyout menu isn't capitalized. An example being "playback speed".

Tbh I think it is more appropriate to treat `Ambient mode` as a proper noun

I have never seen the word `Ambient mode` used by any other platform other than Google's platform ([Android](https://developer.android.com/training/tv/playback/ambient-mode?hl=en), YouTube, YouTube TV)

inotia00 committed:
Apply code review suggestions

KobeW50 commented:
> > Also, "only" is useful because it implies that the `Disable ambient mode` will disable all ambient mode.
> > The string also had "only" before this PR so I think it should be kept.
> 
> Now here's an example. Here also "only in home feed." could be used. But it isn't necessary as there's only one home feed. The case is same here. There's only one fullscreen (whether portrait/landscape as the toggle affects both), it isn't necessary to specify "only in fullscreen" here either.
> 
> ![Screenshot_20240620_235355_YouTube](https://private-user-images.githubusercontent.com/117499019/341522701-f1382456-bb00-47ac-9b79-961da9ad1ed0.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5OTUyMzksIm5iZiI6MTcxODk5NDkzOSwicGF0aCI6Ii8xMTc0OTkwMTkvMzQxNTIyNzAxLWYxMzgyNDU2LWJiMDAtNDdhYy05Yjc5LTk2MWRhOWFkMWVkMC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyMVQxODM1MzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lM2M1MzI1N2RhM2RjZTJkMmQ1YjJhMjU4M2U5ZjU0YjJiNjU2YTc5M2FhNTJmYmZkMjJkZGJiMGYwOTQ4MDgxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.x52i4Nro-5R5gWP6Nt9jDPmWKsA1ptSVE8qWTBIpm8E)

For that case, it is more obvious that the comments section under the video player is entirely separate from the comments section in the feed. In fact, even the `Hide comments section` toggle doesn't hide the comments section in the feed. So adding "only" doesn't serve any benefit.

But for ambient mode, it is reasonable for users to think that ambient mode in fullscreen is highly correlated to ambient mode below the player. The fact that the `Disable ambient mode` toggle disables ambient mode in both places is evidence of this correlation. So "only" is used to explicitly separate them.

ILoveOpenSourceApplications commented:
>The fact that the `Disable ambient mode` toggle disables ambient mode in both places is evidence of this correlation. 

Yes, which is why it's simply `Disable Ambient mode` and the other is `Disable Ambient mode in fullscreen`.

>So "only" is used to explicitly separate them.

Why though? When the option has "fullscreen" written in both option and it's description, why say "only". There's no role for "only" imo as everything is clear even without the "only" part.

KobeW50 commented:
_sigh_ whatever you say :)

It's arbitrary at the end of the day.

KobeW50 committed:
Apply suggestion from review

KobeW50 committed:
Merge branch 'proper-nouns' of https://github.com/KobeW50/inotia00-patches into proper-nouns

inotia00 commented:
Is it okay to merge?

inotia00 commented:
I have to release a new patch, so I'll merge the changes so far

KobeW50 commented:
Sorry, i wasn't able to respond. Yeah, there aren't any more changes to be made regarding this PR
